dashboard/app/models/user.rb

```diff
@@ -968,6 +969,12 @@ def username_required?
     manual? || username_changed?
   end
 
+  private def check_profanity
```

Comment:
Since this function specifically validates the username field, can we call it specifically check_username_profanity, or make the function generic so it can be passed a property name and validate that property for profanity? Minor naming nitpick, but the user model is messy enough that a random validation for `:check_profanity` doesn't tell the reader enough.